packages/bulk-edit/src/browser/bulk-edit-tree/bulk-edit-tree-widget.tsx

```diff
@@ -0,0 +1,230 @@
+                            {afterMatch}
+                        </div>
+                    </div>;
+                }
+            }
+        }
+    }
+
+    protected decorateBulkEditInfoNode(node: BulkEditInfoNode): React.ReactNode {
+        const icon = this.toNodeIcon(node);
+        const name = this.toNodeName(node);
+        const description = this.toNodeDescription(node);
+        const path = this.labelProvider.getLongName(node.uri.withScheme('bulkedit'));
+        return <div title={path} className='bulkEditInfoNode'>
+            {icon && <div className={icon + ' file-icon'}></div>}
+            <div className='name'>{name}</div>
+            <div className='path'>{description}</div>
+        </div>;
+    }
+
+    private async getFileContentsMap(workspaceEdit: monaco.languages.WorkspaceEdit): Promise<Map<string, string>> {
+        const fileContentMap = new Map<string, string>();
+
+        if (workspaceEdit?.edits) {
+            for (const element of workspaceEdit.edits) {
+                if (element) {
+                    const filePath = (('newUri' in element) && element?.newUri?.path) || (('resource' in element) && element?.resource?.path);
+
+                    if (filePath && !fileContentMap.has(filePath)) {
+                        const fileUri = new URI(filePath).withScheme('file');
+                        const resource = await this.fileResourceResolver.resolve(fileUri);
+                        fileContentMap.set(filePath, await resource.readContents());
+                    }
+                }
+            }
+        }
+        return fileContentMap;
+    }
+
+    private async doOpen(node: BulkEditNode): Promise<void> {
+        if (node && node.parent && node.bulkEdit && ('edit' in node.bulkEdit)) {
+            const resultNode = node.parent as BulkEditInfoNode;
+            const leftUri = node.uri;
+            const rightUri = await this.createReplacePreview(resultNode);
+            const diffUri = DiffUris.encode(leftUri, rightUri);
+            const editorWidget = await this.editorManager.open(diffUri, this.getEditorOptions(node));
+            this.editorWidgets.push(editorWidget);
+        }
+    }
+
+    private async createReplacePreview(bulkEditInfoNode: BulkEditInfoNode): Promise<URI> {
+        const fileUri = bulkEditInfoNode.uri;
+        let lines: string[] = [];
+        if (bulkEditInfoNode?.fileContents) {
+            lines = bulkEditInfoNode.fileContents.split('\n');
+            bulkEditInfoNode.children.map((node: BulkEditNode) => {
+                if (node.bulkEdit && ('edit' in node.bulkEdit)) {
+                    const startLineNum = +node.bulkEdit.edit.range.startLineNumber;
+                    if (lines.length > startLineNum) {
+                        const startColumn = +node.bulkEdit.edit.range.startColumn;
+                        const endColumn = +node.bulkEdit.edit.range.endColumn;
```

Comment:
Minor: why is the `+` cast necessary? `startLineNumber`, `startColumn` and `endColumn` are numbers already?
In addition, we are not doing it in other areas such as:

```ts
selection: {
    start: {
        line: node.bulkEdit.edit.range.startLineNumber - 1,
        character: node.bulkEdit.edit.range.startColumn - 1
    },
    end: {
        line: node.bulkEdit.edit.range.endLineNumber - 1,
        character: node.bulkEdit.edit.range.endColumn - 1
    }
}
```

Reply:
Fixed. Thanks